code/datums/beam.dm

```diff
@@ -122,15 +126,20 @@
 	mouse_opacity = 0
 	anchored = 1
 	var/datum/beam/owner
+	use_fade = 0
 
 
 /obj/effect/ebeam/Destroy()
 	owner = null
 	return ..()
 
+/obj/effect/ebeam/New(alpha_fade = 0, time_exist = 10)
+	if(alpha_fade)
```

Comment:
While debugging your delicious code while porting, I figured out why the beams aren't straight.
You want Draw() to run first. So place a spawn(1) before the if.

Reply:
Also place the ..() before the if(alpha_fade) so it doesnt randomly dissapear and shit